source/Applications/Advanced/MaskPointCloud/MaskPointCloud.cpp

```diff
@@ -0,0 +1,207 @@
+
+        std::cout << "Masking point cloud" << std::endl;
+
+        // Copying data points within the mask. Rest is set NaN
+        for (int i = 0; i < height; i++)
+        {
+            for (int j = 0; j < width; j++)
+            {
+                if (mask.at<uint8_t>(i, j) > 0) 
+                {
+                    maskedPointCloud(i, j).r = data(i, j).color.r;
+                    maskedPointCloud(i, j).g = data(i, j).color.g;
+                    maskedPointCloud(i, j).b = data(i, j).color.b;
+                    maskedPointCloud(i, j).x = data(i, j).point.x;
+                    maskedPointCloud(i, j).y = data(i, j).point.y;
+                    maskedPointCloud(i, j).z = data(i, j).point.z;
+                }
+                else 
+                {
+                    maskedPointCloud(i, j).r = bad_point;
+                    maskedPointCloud(i, j).g = bad_point;
+                    maskedPointCloud(i, j).b = bad_point;
+                    maskedPointCloud(i, j).x = bad_point;
+                    maskedPointCloud(i, j).y = bad_point;
+                    maskedPointCloud(i, j).z = bad_point;
+                }
+            }
+        }
+        return maskedPointCloud;
+    }
+
+    void visualizeDepthMap(const pcl::PointCloud<pcl::PointXYZRGB>& pointCloud) 
+    {
+        std::cout << "Converting to Depth map in OpenCV format" << std::endl;
+        cv::Mat zJetColorMap = pointCloudToCvZ(pointCloud);
+        std::cout << "Visualizing Depth map" << std::endl;
+        cv::namedWindow("Depth map", cv::WINDOW_AUTOSIZE);
+        cv::imshow("Depth map", zJetColorMap);
+        cv::waitKey(0);
+    }
+
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        // Reading ZDF-file
+        std::string ZDFFileName = std::string(ZIVID_SAMPLE_DATA_DIR) + "/MiscObjects.zdf";
+        std::cout << "Reading ZDF frame from file: " << ZDFFileName << std::endl;
+        const auto frame = Zivid::Frame(ZDFFileName);
+        const auto pointCloud = frame.pointCloud();
+
+        // Creating mask
+        const int pixelsToDisplay = 300;
+        std::cout << "Generating binary mask of central " << pixelsToDisplay << " x " << pixelsToDisplay << "pixels." << std::endl;
+        const int height = pointCloud.height();
+        const int width = pointCloud.width();
+        const int heightMin = (height - pixelsToDisplay) / 2;
+        const int heightMax = (height + pixelsToDisplay) / 2;
+        const int widthMin = (width - pixelsToDisplay) / 2;
+        const int widthMax = (width + pixelsToDisplay) / 2;
+
+        cv::Mat mask = cv::Mat::zeros(height, width, CV_8U);
+        cv::rectangle(mask, cv::Point(widthMin, heightMin), cv::Point(widthMax, heightMax), cv::Scalar(255, 255, 255), -1);
+       
+        // Creating point cloud structure
+        pcl::PointCloud<pcl::PointXYZRGB> maskedPointCloud;
+
+        // Masking point cloud
+        maskedPointCloud = maskPointCloud(pointCloud, mask);
```

Comment:
I would group these lines of code, delete the first comment, convert the second comment into a printout and place it at the top.